substrate/primitives/genesis-builder/src/lib.rs

```diff
@@ -19,39 +19,92 @@
 
 //! Substrate genesis config builder
 //!
-//! This Runtime API allows to construct `RuntimeGenesisConfig`, in particular:
-//! - serialize the runtime default `RuntimeGenesisConfig` struct into json format,
-//! - put the RuntimeGenesisConfig struct into the storage. Internally this operation calls
-//!   `GenesisBuild::build` function for all runtime pallets, which is typically provided by
-//!   pallet's author.
-//! - deserialize the `RuntimeGenesisConfig` from given json blob and put `RuntimeGenesisConfig`
-//!   into the state storage. Allows to build customized configuration.
+//! This module provides means to interact with `RuntimeGenesisConfig`. Runtime provides a default
```

Comment:
a holistic question: is this crate only capable of providing all of its functionality to non-FRAME-based runtimes? 
If no, we can move it to FRAME and possibly make life easier. 
If yes, it might be worth explaining that `RuntimeGenesisConfig` is the name that FRAME-based runtimes have for their Genesis struct and it is not a "magic" word. 

Not an important point, and ofc we don't really have FRAME alternatives, but is is good to think about this :)

Reply:
It can be even trickier. In runtime interface description we shall not refer to any implementation details.
The `Vec<u8>` does not necessarily need to be a JSON blob. If we want to keep the doc super-generic and non opinionated we should rewrite it, and make a special section how is it implemented for FRAME?

Not sure what approach should we take here.

Reply:
92e6824b1a2